### PR TITLE
fix(platform): preserves scrolling position

### LINKED ISSUE #10024: [Sourcing] Tree table: Ability to restore to last table state when user navigates back and forth between pages using Angular router
Is this a bug, enhancement, or feature request?
--
Feature request

Briefly describe your proposal.
--
Whenever user performs any type of editing to the row data on a new page (navigated via Angular router), the table view should retain its last viewed page when user navigates back from that editing page view. 

However, as of now, even with the fixes done in #9907, the `<fdp-table>` component instance gets destroyed when we navigated away and then came back. Due to this, we are unable to use those newly added functionalities to restore to previous table state.

Which versions of Angular and Fundamental Library for Angular are affected? (If this is a feature request, use current version.)
--
Current release.

### PR DESCRIPTION
**Pull request by fkolar**
## Related Issue(s)

closes #10024

## Description

Introduces new TableState property in order to capture the vertical scrolling position called `scrollTopPosition`.
This PR  also updates one part inside the `_listenToPageScrolling` to keep it consistent so we access only public API (getScrollTop) and not directly to its elementRef property members

##### PR Quality

-   [x] the commit message(s) follows the guideline:
        https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
-   [x] tests for the changes that have been done
-   [x] all items on the PR Review Checklist are addressed :
        https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

### CODE REVIEW

File: libs/docs/platform/table/examples/virtual-scroll/platform-table-virtual-scroll-example.component.ts [hunk 1]
```diff
@@ -16,12 +17,38 @@ import {
...
+                category: null,
+                text: ''
+            },
+            freezeToColumn: null,
+            freezeToEndColumn: null,
+            page: {
+                pageSize: 0,
+                currentPage: 1
+            },
+            scrollTopPosition: 3310
```
Comment (InnaAtanasova) [inline libs/docs/platform/table/examples/virtual-scroll/platform-table-virtual-scroll-example.component.ts:49]: how do you determine this value?

File: libs/platform/src/lib/table-helpers/directives/table-virtual-scroll.directive.ts [hunk 2]
```diff
@@ -129,8 +129,9 @@ export class TableVirtualScrollDirective extends TableVirtualScroll implements O
                 filter(() => this.virtualScroll && !!this.bodyHeight),
                 takeUntil(this._destroy$)
             )
-            .subscribe(() => {
+            .subscribe((scrollable) => {
                 this.calculateVirtualScrollRows();
+                this._table.getTableState().scrollTopPosition = scrollable.getScrollTop();
```
Comment (N1XUS) [inline libs/platform/src/lib/table-helpers/directives/table-virtual-scroll.directive.ts:134]: use `setTableState` instead

File: libs/platform/src/lib/table/table.component.ts [hunk 3]
```diff
@@ -1963,6 +1967,17 @@ export class TableComponent<T = any>
         });
     }
 
+    /** @hidden */
+    private _initScrollPosition(): void {
+        const state = this.getTableState();
+
+        console.log(`_initScrollPosition: ${state.scrollTopPosition}  - ${this._tableRows.length}`);
```
Comment (N1XUS) [inline libs/platform/src/lib/table/table.component.ts:1974]: Remove logging

File: libs/docs/platform/table/examples/virtual-scroll/platform-table-virtual-scroll-example.component.ts (hunk 1 above)
Comment (mikerodonnell89) [reply libs/docs/platform/table/examples/virtual-scroll/platform-table-virtual-scroll-example.component.ts:49]: arbitrary for example purposes i am guessing

Comment (fkolar) [reply libs/docs/platform/table/examples/virtual-scroll/platform-table-virtual-scroll-example.component.ts:49]: This is just for the exampla to showcase that, you can pre-initialize a table with some initial scroll position that you might have saved. 
The value was taken from this specific example, where I wanted this table to scroll to.

File: libs/platform/src/lib/table-helpers/directives/table-virtual-scroll.directive.ts (hunk 2 above)
Comment (fkolar) [reply libs/platform/src/lib/table-helpers/directives/table-virtual-scroll.directive.ts:134]: ok. I assume you mean use setTableState and set this value using spread operator.

File: libs/platform/src/lib/table/table.component.ts (hunk 3 above)
Comment (fkolar) [reply libs/platform/src/lib/table/table.component.ts:1974]: sure thing

**Review by g-cheishvili — APPROVED** [review]

**Review by InnaAtanasova — APPROVED** [review]

**Pull request merged**